src/Plugins/ComplexCore/src/ComplexCore/Filters/CopyFeatureArrayToElementArray.cpp

```diff
@@ -0,0 +1,202 @@
+Uuid CopyFeatureArrayToElementArray::uuid() const
+{
+  return FilterTraits<CopyFeatureArrayToElementArray>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string CopyFeatureArrayToElementArray::humanName() const
+{
+  return "Create Element Array from Feature Array";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> CopyFeatureArrayToElementArray::defaultTags() const
+{
+  return {"#Core", "#Memory Management"};
+}
+
+//------------------------------------------------------------------------------
+Parameters CopyFeatureArrayToElementArray::parameters() const
+{
+  Parameters params;
+  // Create the parameter descriptors that are needed for this filter
+  //  params.insertSeparator(Parameters::Separator{"Feature Data"});
+  params.insert(std::make_unique<ArraySelectionParameter>(k_SelectedFeatureArrayPath_Key, "Feature Data to Copy to Element Data", "", DataPath{}));
+  //  params.insertSeparator(Parameters::Separator{"Element Data"});
+  params.insert(std::make_unique<ArraySelectionParameter>(k_FeatureIdsArrayPath_Key, "Feature Ids", "", DataPath{}, false, ArraySelectionParameter::AllowedTypes{DataType::int32}));
+  //  params.insertSeparator(Parameters::Separator{"Element Data"});
+  params.insert(std::make_unique<ArrayCreationParameter>(k_CreatedArrayName_Key, "Copied Attribute Array", "", DataPath{}));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer CopyFeatureArrayToElementArray::clone() const
+{
+  return std::make_unique<CopyFeatureArrayToElementArray>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult CopyFeatureArrayToElementArray::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                                       const std::atomic_bool& shouldCancel) const
+{
+  auto pSelectedFeatureArrayPathValue = filterArgs.value<DataPath>(k_SelectedFeatureArrayPath_Key);
+  auto pFeatureIdsArrayPathValue = filterArgs.value<DataPath>(k_FeatureIdsArrayPath_Key);
+  auto pCreatedArrayNameValue = filterArgs.value<DataPath>(k_CreatedArrayName_Key);
+
+  const IDataArray& selectedFeatureArray = dataStructure.getDataRefAs<IDataArray>(pSelectedFeatureArrayPathValue);
+  const IDataStore* selectedFeatureArrayStore = selectedFeatureArray.getIDataStore();
+  const IDataArray& featureIdsArray = dataStructure.getDataRefAs<IDataArray>(pFeatureIdsArrayPathValue);
+  const IDataStore* featureIdsArrayStore = featureIdsArray.getIDataStore();
+
+  complex::Result<OutputActions> resultOutputActions;
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  DataType dataType = selectedFeatureArrayStore->getDataType();
+  NumericType numericType = static_cast<NumericType>(dataType);
```

Comment:
We should probably unify these two enums though that'll be a different PR. For this PR, a dedicated function would be better since a simple static_cast is unchecked and prone to error.